src/app/modules/category/components/filter/filter.component.ts

```diff
@@ -1,12 +1,110 @@
-import { Component, OnInit } from '@angular/core';
+import {
+    Component,
+    OnDestroy,
+    OnInit,
+    ChangeDetectionStrategy,
+} from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subject } from 'rxjs';
+import { filter, map, takeUntil } from 'rxjs/operators';
+import { ShareService } from 'src/app/services/share/share.service';
+import { AddToArray } from '../../state/category.models';
+import * as CategorySelectors from '../../state/selectors/category.selectors';
 
 @Component({
     selector: 'app-filter',
     templateUrl: './filter.component.html',
     styleUrls: ['./filter.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class FilterComponent implements OnInit {
-    constructor() {}
+export class FilterComponent implements OnInit, OnDestroy {
+    private readonly unsubscribe$ = new Subject();
+
+    array: AddToArray[] = [];
+    private filterItem: AddToArray;
+    inputValue1 = '';
+    inputValue2 = '';
+    selectValue1 = '';
+    selectValue2 = '';
+
+    readonly selectFilterData$ = this.store
+        .select(CategorySelectors.selectFilterData)
+        .pipe(
+            filter(el => el !== undefined),
+            map(el => el),
+            takeUntil(this.unsubscribe$),
+        );
+
+    readonly isReadyToDisplay$ = this.selectFilterData$.pipe(
+        map(el => !!el),
+        takeUntil(this.unsubscribe$),
```

Comment:
we should not duplicate takeUntil(this.unsubscribe$) more then once, u have set it for selectFilterData$, so here it is redundant

Reply:
Ok, thank you. But what about combineLatest. If I use two observables in it, thats mean that I can use just one takeUntil(this.unsubscribe$) in one of two observables ?

Reply:
i believe you should set takeUntil for result combined observable or to all observables that was passed to the ombinelatest